### ISSUE TITLE
CMakeLists doesn't build a library

### ISSUE THREAD

**Issue by oedze** [debevv/nanoMODBUS#72]
The CMakeLists is now used to create a few executables with different settings.

It can't however be used as a static library right now which is usefull for embedded devices.

being able to do 

```txt
add_subdirectory(path/to/nanoMODBUS)

add_executable(myproject main.c)

target_link_libaries(myproject PRIVATE nanomodbus)
```

in our main projects CMakeLists.txt would be ideal for our scenario.

To be able to do this requires some change to the current CMakeLists.txt. If anyone's interested i can make a pull request.
If i do this i would also prefer to move the nanomodbus.h to a include folder.

**Comment by donghoonpark** [comment]
How about supporting both approaches?
Adding ```nanomodbus``` as a library in CMake and compiling test examples with the library link would be the easiest way to achieve this.
Furthermore, for the project's CI, adopting a unit testing framework is also worth considering. If you don’t have time to implement it, I’d be happy to create a PR for it. Is there any particular reason you don’t want to make it a CMake-integrable library?

**Comment by debevv** [comment]
No reason really, I just think that dropping the sources in the codebase is the simplest option. Anyway, if you find it useful, I will add a couple of lines in the CMakeLists to build a library.  
Regarding the tests, what do you mean with a unit testing framework?

**Comment by debevv** [comment]
Added library target in a1705c0dff8e376c3ff45c73330bb3f1dbae2c18

**Comment by donghoonpark** [comment]
I really appreciate your quick feedback. 
Library definition is quite useful when I use library with internet loading cmake command such as ```FetchContent```.
Regarding the unittest framework, to build test cases with minimal code and integrate them easily into the development environment, frameworks such as GoogleTest or Catch2 are commonly used, as they offer efficiency and compatibility.

**Comment by donghoonpark** [comment]
```nanomodbus``` have no extra library. If you call ```add_subdirectory```, it gives you ```nanomodbus``` library spec. I already made a PR #79 that includes header file in library spec as well. Until the PR merge into master, you can refer the example [HERE](https://github.com/debevv/nanoMODBUS/blob/master/examples/stm32/CMakeLists.txt#L55).

**Comment by patrick-dojofive** [comment]
#79 is merged so is this issue resolved?

**Comment by debevv** [comment]
Yes, both static and dynamic libraries can be built, look at [CMakeLists.txt:11](https://github.com/debevv/nanoMODBUS/blob/a980f7f4c427e75203a740e84e30503edccd3e1e/CMakeLists.txt#L11C10-L11C27) for reference
